### PR TITLE
Team 5 v2

### PR DESCRIPTION
**Pull request by jzzheng22**
# Summary
Team 5 MVP 2.0

FYI have also commented out some annoying and/or duplicate logs in team1agent1 and simenv

### CODE REVIEW

**Review by hussain2603 — DISMISSED** [review]
looks fine to me

File: pkg/agents/team5/agent.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+		a.dayPassed()
+	}
+
+	a.getMessages()
+	a.dailyMessages()
+
+	//When platform reaches our floor and we haven't tried to eat, then try to eat
+	if a.CurrPlatFood() != -1 && !a.HasEaten() {
+		lastMeal, err := a.TakeFood(a.attemptFood)
+		if err != nil {
```
Comment (JaafarRammal) [inline pkg/agents/team5/agent.go:123]: I see you logging errors to the default console logger here. These will never go to a logger file and we intended to use the console logger for the backend monitoring. Two propositions:
- Add an ID (e.g. "Simulation - {FILENAME}") next to it so we can identify it in the backend. You can even comment it out and I can deal with auto-id for these type of logs
- Use your agent logger, can add a log.Error to the base agent logging

File: pkg/agents/team5/agent.go [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+		surroundingAgents: make(map[int]uuid.UUID),    // Map agent IDs of surrounding floors relative to current floor
+	}, nil
+}
+
+func (a *CustomAgent5) updateAimHP() {
+	a.currentAimHP = a.HealthInfo().MaxHP - ((10 - a.selfishness) * ((a.HealthInfo().MaxHP - a.HealthInfo().WeakLevel) / 10))
+}
+
+func (a *CustomAgent5) updateSelfishness() {
+	a.selfishness = 10 - a.calculateAverageFavour()
```
Comment (JaafarRammal) [inline pkg/agents/team5/agent.go:59]: I see you intended for selfishness to be 0 to 10, would this ensure it's never out of range?

**Review by JaafarRammal — COMMENTED** [review]
Overall good architecture and separation of functions. A few comments about state check for accesses and values

File: pkg/simulation/simIteration.go [hunk 3]
```diff
@@ -26,7 +26,7 @@ func (sE *SimEnv) simulationLoop(t *infra.Tower) {
 		sE.AgentsRun(t)
 		sE.TowerTick()
 		sE.replaceAgents(t)
-		t.TowerStateLog(" end of tick")
+		// t.TowerStateLog(" end of tick")
```
[the remark this answers is not in the document]
Comment (jzzheng22) [reply pkg/simulation/simIteration.go:29]: I commented out deliberately - end of tick and start of tick are the same so no point in logging twice. Can I just delete the line entirely?

File: pkg/agents/team5/utils.go [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+	return num
+}
+
+func (a *CustomAgent5) memoryIdExists(id uuid.UUID) bool {
+	_, exists := a.socialMemory[id]
+	return exists
+}
+
+func (a *CustomAgent5) ResetSurroundingAgents() {
+	a.surroundingAgents = make(map[int]uuid.UUID)
```
[the remark this answers is not in the document]
Comment (jzzheng22) [reply pkg/agents/team5/utils.go:25]: Garbage collected language, should be fine (I hope...)

File: pkg/agents/team5/agent.go (hunk 2 above)
Comment (jzzheng22) [reply pkg/agents/team5/agent.go:59]: Yes, favour is bounded between 0 and 10 inclusive, so calculateAverageFavour can't take any values outside of this range.

File: pkg/agents/team5/social.go [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+		foodTaken:         100,
+		agentHP:           a.HealthInfo().MaxHP,
+		intentionFood:     100,
+		favour:            0,
+		daysSinceLastSeen: 0,
+	}
+}
+
+func (a *CustomAgent5) updateFoodTakenMemory(id uuid.UUID, foodTaken food.FoodType) {
+	mem := a.socialMemory[id]
```
[the remark this answers is not in the document]
Comment (jzzheng22) [reply pkg/agents/team5/social.go:21]: We call the exists function before the call to this function

**Review by hussain2603 — APPROVED** [review]

**Pull request merged**